packages/flutter/lib/src/services/system_chrome.dart

```diff
@@ -368,6 +368,68 @@ abstract final class SystemChrome {
   ///
   /// ## Limitations
   ///
+  /// ### Android
+  ///
+  /// Android screens may choose to [letterbox](https://developer.android.com/guide/practices/enhanced-letterboxing)
+  /// applications that lock orientation, particularly on larger screens. When
+  /// letterboxing occurs on Android, the [MediaQueryData.size] reports the
+  /// letterboxed size, not the full screen size. Applications that make
+  /// decisions about whether to lock orientation based on the screen size
+  /// must use the `display` property of the current [FlutterView].
+  ///
+  /// ```dart
+  /// // A widget that locks the screen to portrait if it is less than 600
+  /// // logical pixels wide.
+  /// class MyApp extends StatefulWidget {
+  ///   @override
+  ///   State<StatefulWidget> createState() => _MyAppState();
+  /// }
+  ///
+  /// class _MyAppState extends State<MyApp> with WidgetsBindingObserver {
+  ///   ui.Display? _display;
+  ///   static const double kOrientationLockBreakpoint = 600;
+  ///
+  ///   @override
+  ///   void didChangeDependencies() {
+  ///     super.didChangeDependencies();
+  ///     _display = View.maybeOf(context)?.display;
+  ///     WidgetsBinding.instance.removeObserver(this);
+  ///     WidgetsBinding.instance.addObserver(this);
```

Comment:
This should only have to happen once in initState?

Reply:
Ahh yes. Then I don't need the remove/add...